src/library_pthread.js

```diff
@@ -832,6 +832,10 @@ var LibraryPThread = {
   pthread_exit: function(status) {
     if (!ENVIRONMENT_IS_PTHREAD) _exit(status);
     else PThread.threadExit(status);
+#if WASM_BACKEND
+    // pthread_exit is marked noReturn, so we must not return from it.
```

Comment:
In the wasm backend specifically, it will emit an unreachable here (maybe just in -O0? I didn't check), so we must not get there (it doesn't make the app fail in most cases, but it is shown in the console as an error).